programs/fileio.c

```diff
@@ -1999,16 +2007,105 @@ int FIO_compressMultipleFilenames(FIO_ctx_t* const fCtx,
 /* **************************************************************************
  *  Decompression
  ***************************************************************************/
+#define DECOMPRESSION_MAX_WRITE_JOBS    (10)
+
+typedef struct {
+    POOL_ctx* writerPool;
+    ZSTD_pthread_mutex_t writeJobsMutex;
+    void* jobs[DECOMPRESSION_MAX_WRITE_JOBS];
+    volatile int availableWriteJobs;
```

Comment:
`availableWriteJobs` does not need to be volatile.

But can you add a newline before `writeJobsMutex` and after `availableWriteJobs`, and a comment explaining that the mutex protects `jobs` and `availableWriteJobs`?